### PR TITLE
Scrape Net Profit and store into a Stock object

### PR DESCRIPTION
**Pull request by ccpandhare**
We might want to use numpy or something else for storing these values.

That will make it easier to do stuff on the fly like profit_margin = net_profit / revenue

### CODE REVIEW

File: lib/scraper.py [hunk 1]
```diff
@@ -1,5 +1,40 @@
...
-def scrape(ticker: str) -> Stock:
-    raise Exception("scraper.scrape: Not implemented")
+# Ticker: pd.DataFrame: Name, Ticker, URL
+def scrape(session: requests.Session, ticker: str) -> Stock:
+    debug("Scraping %s" % ticker)
+    url = "https://www.screener.in/company/%s/consolidated/" % ticker
+    page = session.get(url)
+    soup = BeautifulSoup(page.text, 'html.parser')
+    name = soup.find('h1').text
+    [net_profit] = parse_table(soup.select_one('#profit-loss .data-table'), ['net profit'])
```
Comment (AshutoshSancheti) [inline lib/scraper.py]: The parse data is in the form of a list of dictionaries. Can we directly add these values into a dataframe with the year & financial metrics being columns? Plus can we add range for years?

File: lib/scraper.py [hunk 2]
```diff
@@ -1,5 +1,40 @@
...
+
+    stock = Stock(ticker, name, url, net_profit)
+    return stock
```
Comment (AshutoshSancheti) [inline lib/scraper.py:26]: Instead of having individual metrics as args, we can declare a dataframe and update that.

**Review by AshutoshSancheti — COMMENTED** [review]
Hi CCD,

I have couple of suggestions on the way we handle data post scraping

File: lib/scraper.py (hunk 1 above)
Comment (ccpandhare) [reply lib/scraper.py]: I think that makes sense - let me do that

File: lib/scraper.py (hunk 2 above)
Comment (ccpandhare) [reply lib/scraper.py:26]: Yeah that seems like the way to go

File: lib/scraper.py (hunk 1 above)
Comment (ccpandhare) [reply lib/scraper.py]: Let me spend a few hours learning dataframes

**Comment by ccpandhare** [comment]
Thanks for the suggestion, @AshutoshSancheti! This is so much cleaner!

**Comment by ccpandhare** [comment]
Screenshot 2022-04-24 at 22 38 15

**Review by AshutoshSancheti — APPROVED** [review]
Nice one, Chinmay.
Going ahead with the merge

**Comment by AshutoshSancheti** [comment]
Queries resolved, changes merged.

**Pull request merged**